scripts/build-types.mjs

```diff
@@ -0,0 +1,31 @@
+/* eslint-disable no-console, no-process-exit, functional/no-expression-statement */
+import path from 'path';
+import * as url from 'url';
+
+import { exec } from './utils/exec.mjs';
+
+const apiExtractorPath = path.resolve(
+  url.fileURLToPath(new URL('.', import.meta.url)), // __dirname es module style
```

Comment:
Since we use this in 3 scripts we could have a `dirname` function in `utils` to abstract it.

```js
function dirname(path) {
  return url.fileURLToPath(new URL(path, import.meta.url));
}
```

Reply:
If that is in the `scripts/utils` directory won't that always resolve to that directory, not the one it's called from?

Reply:
Ah yes :facepalm: We can keep it that way, I'll suggest local edits for consistency.

Reply:
I totally agree it's ugly and not self explanatory of what is happening, that is why I had a comment.